src/wix/WixToolset.Core.Burn/Bundles/BurnWriter.cs

```diff
@@ -138,10 +143,19 @@ public bool AppendContainer(Stream containerStream, long containerSize, BurnComm
                     break;
 
                 case Container.Attached:
-                    burnSectionCount = 2;
-                    burnSectionOffsetSize = BURN_SECTION_OFFSET_ATTACHEDCONTAINERSIZE;
+                    burnSectionCount = 2 + (uint)this.AttachedContainers.Count;
+                    burnSectionOffsetSize = BURN_SECTION_OFFSET_ATTACHEDCONTAINERSIZE0 + ((uint)this.AttachedContainers.Count * 4);
```

Comment:
`burnSectionCount` is a bundle level value. I guess we've been setting it multiple times all along so this doesn't really matter, but it's unnecessary to do it for every container.

`burnSectionOffsetSize` is a container level value that is an offset to the location of its size. `this.AttachedContainers.Count` needs to be replaced with the container's index so each container's size is written to a different location.

Reply:
AttachedContainers.Count is incremented on each call, so both values should be updated